### ISSUE TITLE
azuredevops_serviceendpoint_aws resource "forget" the secret key if the role_to_assume changes

### ISSUE THREAD

**Issue by themaroqa** [microsoft/terraform-provider-azuredevops#732]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and Azure DevOps Provider) Version

terraform version 1.3.8
azuredevops provider version 0.3.0

### Affected Resource(s)

* `azuredevops_serviceendpoint_aws`

### Terraform Configuration Files

```hcl
data "azuredevops_project" "example" {
  name = "project_name"
}

resource "azuredevops_serviceendpoint_aws" "example" {
  project_id            = azuredevops_project.example.id
  service_endpoint_name = "Example AWS"
  access_key_id         = "00000000-0000-0000-0000-000000000000"
  secret_access_key     = "accesskey"
  role_to_assume        = "arn:aws:iam::00000000000:role/role_name"
  description           = "Managed by AzureDevOps"
}
```

### Steps to Reproduce

1. The service endpoint "Example AWS" exists and works
2. Change the role_to_assume value to "arn:aws:iam::00000000000:role/role_name_changed" (the role should exists on AWS)
3. `terraform apply`
4. The apply completes succesfully, but then the service endpoint in Azure DevOps stops working. It seems it tries to connect to AWS without a secret_access_key value passed. The error I got from AWS is `##[error]SignatureDoesNotMatch: The request signature we calculated does not match the signature you provided. Check your AWS Secret Access Key and signing method. Consult the service documentation for details.`
5. Re-adding manually the same secret key directly in the Azure DevOps fixes the issue.

**Comment by davidcorrigan714** [comment]
Essentially a dup of #719, got a branch with a few fixes about ready for a PR and can fix this with the rest of them. It's all the same root cause.

**Comment by themaroqa** [comment]
fix released on v0.8.0
